### PR TITLE
cairo: 1.16.0 -> 1.18.0

### PR DESCRIPTION
**Pull request by alyssais**
## Hydra comparisons

- [1800235](https://hydra.nixos.org/eval/1800235?compare=1799588&full=0) (complete)

## Description of changes

 - libiconv and expat are no longer used.
 - libintl and xcbutil no longer appear to be used.
 - The GL backend has been removed.
 - We can no longer disable PDF support, since something always propagates zlib.
 - Not sure whether the Requires.private is still needed, so I've left it.

Fixes #256519.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by RaitoBezarius** [comment]
Do you want to get a Hydra jobset for this?

**Comment by bobby285271** [comment]
The only build fixes for this in GNOME 45 PR so far are 2968007ad4786fd9a3ea731a7b91f31ac5f282cb and cd6c696b0dbc03961c8f6127c767c18007803fb8 if you wish to do this in separate PR (but the GNOME 45 does need the bump for librsvg, glycin-loaders, loupe, etc)

**Comment by alyssais** [comment]
@vcunat could we get a Hydra jobset?

**Comment by bobby285271** [comment]
(The GNOME 45 PR now also have 517d24e6accda409b0786d317619027e022d1354, ed5e1e18cc13416e96244a6978277a3a43684459, and I-don't-know-why 12b140ca8d9bd7762c3e2f86a608a42a83f3ae51)

**Comment by vcunat** [comment]
Jobset created: https://hydra.nixos.org/jobset/nixpkgs/pr-257301-cairo-1.18

I doubt that it will be possible to get significant amounts of builds for this PR done this week, with all-rebuilding rush starting on Wednesday due to security fixes.  I guess we'll also wait with gnome jobset until this progresses?  (it also contains the update and it doesn't make much sense to me for these jobsets to compete against each other for builders)

**Comment by vcunat** [comment]
I triggered it for x86_64-linux and aarch64-darwin for now.

**Comment by vcunat** [comment]
Builds finished, FYI.

**Comment by vcunat** [comment]
Uh, the push... do we really need to rebuild 100k builds now?

**Comment by alyssais** [comment]
No, go ahead and cancel it.

**Comment by vcunat** [comment]
I had one eval per day timer in there, but I triggered this one myself :-)  I thought it was mainly the two extra commits shown by GitHub UI.

**Comment by vcunat** [comment]
So I suppose you'll soon consider this ready to merge, without another Hydra eval?  There are only 10 days left to merge breaking changes that should reach NixOS 23.11.

**Comment by vcunat** [comment]
OK, I stopped the eval timer.  Ping me when you want it.

**Comment by alyssais** [comment]
@bobby285271 I've fixed cogl in a different way, just linking libGL and avoiding the need for a patch.

**Comment by alyssais** [comment]
@vcunat I've now addressed all the direct new build failures from the last Hydra run, so I think it's time for another.

**Review by fabianhjr — APPROVED** [review]
Been using cairo 1.18.0 as part of GNOME 45 branch: https://github.com/NixOS/nixpkgs/pull/247766/files#diff-c1b6675856a14491b8c610758d37ad4372e04a3f25a6049926c738f21121e5b0

Eval looks great with minimal breakage ( https://hydra.nixos.org/eval/1800745 )

**Comment by vcunat** [comment]
After taking various things into consideration, let's finish this during the upcoming `staging-next`.

**Comment by bobby285271** [comment]
And we probably still need to fix cross there I guess (did not test, but I saw the `ipc_rmid_deferred_release` lines commented) :thinking:

**Comment by alyssais** [comment]
Oh, whoops.

**Comment by trofi** [comment]
I noticed that `cairo` bump changed `x11Support` condition from `!stdenv.isDarwin` to `!stdenv.isDarwin || true` which seems to be an effective `true`. Should it be just `true`? Or it's a debug leftover that should be `!stdenv.isDarwin`?

File: pkgs/development/libraries/cairo/default.nix [hunk 1]
```diff
@@ -1,12 +1,8 @@
-{ config, lib, stdenv, fetchurl, fetchpatch, pkg-config, libiconv
-, libintl, expat, zlib, libpng, pixman, fontconfig, freetype
-, x11Support? !stdenv.isDarwin, libXext, libXrender
+{ lib, stdenv, fetchurl, fetchpatch, gtk-doc, meson, ninja, pkg-config, python3
+, docbook_xsl, fontconfig, freetype, libpng, pixman, zlib
+, x11Support? !stdenv.isDarwin || true, libXext, libXrender
```
Comment (fabianhjr) [inline pkgs/development/libraries/cairo/default.nix:3]: Good observation, my instinct thinks this is an unintended change and `|| true` should be removed but tagging @NixOS/darwin-core for decision on disabling or enabling on darwin by default.

**Comment by fabianhjr** [comment]
Mhhh, and this is a year old. Probably best to leave enabled then? (It has been part of the last 2 stable releases)

**Comment by reckenrode** [comment]
I would prefer not to pull in X11 libraries unnecessarily, but what’s the effect on closure size?

**Pull request merged**